src/commands/install-eksctl.yml

```diff
@@ -6,18 +6,28 @@ steps:
   - run:
       name: Install the eksctl tool
       command: |
-        if which eksctl > /dev/null; then
-          echo "eksctl is already installed"
-          exit 0
+        if [[ $EUID == 0 ]]; then export SUDO=""; else export SUDO="sudo"; fi
+        # Define current platform
+        if [[ "$(uname -s)" == "Darwin" && "$(uname -m)" == "x86_64" ]]; then
+          export SYS_ENV_PLATFORM=macos
+        elif [[ "$(uname -s)" == "Linux" && "$(uname -m)" == "x86_64" ]]; then
+          export SYS_ENV_PLATFORM=linux_x86
+        elif [[ "$(uname -s)" == "Linux" && "$(uname -m)" == "aarch64" ]]; then
+          export SYS_ENV_PLATFORM=linux_arm
+        else
+          echo "This platform appears to be unsupported."
+          uname -a
+          exit 1
         fi
 
-        mkdir -p eksctl_download
-        curl --silent --location --retry 5 "https://github.com/weaveworks/eksctl/releases/download/latest_release/eksctl_$(uname -s)_amd64.tar.gz" \
-          | tar xz -C eksctl_download
-        chmod +x eksctl_download/eksctl
-        SUDO=""
-        if [ $(id -u) -ne 0 ] && which sudo > /dev/null ; then
-          SUDO="sudo"
+        if ! command -v eksctl >/dev/null 2>&1; then
+          echo "Installing the AWS eksctl CLI"
+            curl --silent --location "https://github.com/weaveworks/eksctl/releases/latest/download/eksctl_$(uname -s)_amd64.tar.gz" | tar xz -C /tmp
```

Comment:
I see we define the current platform in 9-20, but we still use uname here. Is this correct?

Reply:
fixed in next commit